packages/grafana-ui/src/components/DataSourceSettings/CustomHeadersSettings.tsx

```diff
@@ -206,6 +206,7 @@ export class CustomHeadersSettings extends PureComponent<Props, State> {
           <Button
             variant="secondary"
             icon="plus"
+            type="button"
```

Comment:
@hugohaggmark would it make sense to make this a default type on our button to prevent this in the future?

Reply:
I agree, wasn't aware of this.

Reply:
Opened #31887 to address this.